### PR TITLE
Matching ROSA day1

### PR DESCRIPTION
**Pull request by bardielle**
This PR includes adding and adapting the attributes in cluster_rosa_classic resource in order to be compatible with ROSA CLI day1 commands for cluster 
    *Change the attribute external_id as an option attribute
    *Change the name of compute_node attribute to be replicas
    *Add tags as a new optional attribute
    *Add DisableWorkloadMonitoring as a new optional attribute
    *Add DisableSCPChecks as a new optional attribute
    *Add FIPS as a new optional attribute
    *Add KMSKeyArn as a new optional attribute

Also, this PR supports a new command in the makefile to generate and update the doc files.

### CODE REVIEW

**Comment by bardielle** [comment]
@oriAdler

File: Makefile [hunk 1]
```diff
@@ -85,3 +85,6 @@ fmt: fmt_go fmt_tf
 .PHONY: clean
 clean:
 	rm -rf .terraform.d
+
+generate:
+	go generate ./...
```
Comment (oriAdler) [inline Makefile:90]: It seems like it missing a new line here at the end of the file.

File: docs/resources/cluster_rosa_classic.md [hunk 2]
```diff
@@ -28,26 +28,31 @@ OpenShift managed cluster using rosa sts.
...
 - `aws_subnet_ids` (List of String) aws subnet ids
 - `compute_machine_type` (String) Identifier of the machine type used by the compute nodes, for example `r5.xlarge`. Use the `ocm_machine_types` data source to find the possible values.
-- `compute_nodes` (Number) Number of compute nodes of the cluster.
+- `disable_scp_checks` (Boolean) Enables you to monitor your own projects in isolation from Red Hat Site Reliability Engineer (SRE) platform metrics.
+- `disable_workload_monitoring` (Boolean) Enables you to monitor your own projects in isolation from Red Hat Site Reliability Engineer (SRE) platform metrics.
 - `etcd_encryption` (Boolean) Encrypt etcd data.
+- `external_id` (String) Unique external identifier of the cluster.
+- `fips` (Boolean) Create cluster that uses FIPS Validated / Modules in Process cryptographic libraries
 - `host_prefix` (Number) Length of the prefix of the subnet assigned to each node.
+- `kms_key_arn` (String) The key ARN is the Amazon Resource Name (ARN) of a CMK. It is a unique, fully qualified identifier for the CMK. A key ARN includes the AWS account, Region, and the key ID.
```
Comment (oriAdler) [inline docs/resources/cluster_rosa_classic.md:37]: What does `CMK` stands for?

File: provider/cluster_rosa_classic_resource.go [hunk 3]
```diff
@@ -326,12 +415,54 @@ func createClassicClusterObject(ctx context.Context,
...
 
 	aws := cmv1.NewAWS()
+
+	if !state.Tags.Unknown && !state.Tags.Null {
+		tags := map[string]string{}
+		for k, v := range state.Tags.Elems {
+			if _, ok := tags[k]; ok {
+				errDescription := fmt.Sprintf("Invalid tags, user tag keys must be unique, duplicate key '%s' found", k)
+				logger.Error(ctx, errDescription)
+				errHeadline := "Can't build cluster"
```
Comment (oriAdler) [inline provider/cluster_rosa_classic_resource.go:432]: We can declare this variable outside of the `if` scope and then reuse it in line `454`.

File: docs/resources/cluster_rosa_classic.md (hunk 2 above)
Comment (bardielle) [reply docs/resources/cluster_rosa_classic.md:37]: I copied it from rosa - https://github.com/openshift/rosa/blob/master/cmd/create/cluster/cmd.go#L342

Comment (gdbranco) [reply docs/resources/cluster_rosa_classic.md:37]: That needs to be update Customer Master Key (CMK) has been replaced with AWS KMS (Key Management Service) Key

File: provider/cluster_rosa_classic_resource.go (hunk 3 above)
Comment (gdbranco) [reply provider/cluster_rosa_classic_resource.go:432]: Could be outside the loop too I suppose

File: provider/cluster_rosa_classic_resource.go [hunk 4]
```diff
@@ -326,12 +415,54 @@ func createClassicClusterObject(ctx context.Context,
...
+
+		aws.Tags(tags)
+	}
+
+	if !state.KMSKeyArn.Unknown && !state.KMSKeyArn.Null && state.KMSKeyArn.Value != "" {
+		kmsKeyARN := state.KMSKeyArn.Value
+		if !kmsArnRE.MatchString(kmsKeyARN) {
+			errDescription := fmt.Sprintf("Expected a valid value for kms-key-arn matching %s", kmsArnRE)
+			logger.Error(ctx, errDescription)
+			errHeadline := "Can't build cluster"
```
Comment (gdbranco) [inline provider/cluster_rosa_classic_resource.go:451]: Same thing from line `432` I suppose

File: provider/cluster_rosa_classic_resource.go [hunk 5]
```diff
@@ -686,13 +821,31 @@ func populateRosaClassicClusterState(ctx context.Context, object *cmv1.Cluster,
...
+	disableUserWorkload, ok := object.GetDisableUserWorkloadMonitoring()
+	if ok && disableUserWorkload {
+		state.DisableWorkloadMonitoring = types.Bool{
+			Value: true,
+		}
+	}
+	getAdditionalTrustBundle, ok := object.GetAdditionalTrustBundle()
+	if ok {
+		// TODO - how to retrieve the file name??
+		logger.Debug(ctx, getAdditionalTrustBundle)
```
Comment (gdbranco) [inline provider/cluster_rosa_classic_resource.go:840]: I think this is only the contents of the file as a string

File: Makefile (hunk 1 above)
Comment (bardielle) [reply Makefile:90]: Nice catch thank you

File: provider/cluster_rosa_classic_resource.go (hunk 5 above)
Comment (bardielle) [reply provider/cluster_rosa_classic_resource.go:840]: I removed that code - @oriAdler and I will decide tomorrow how to deal with additional trust bundle

**Review by gdbranco — APPROVED** [review]
lgtm

**Pull request merged**